src/bp-search/classes/class-bp-search-posts.php

```diff
@@ -60,90 +83,91 @@ public function sql( $search_term, $only_totalrow_count = false ) {
 			*/
 			global $wpdb;
 
-			$bp_prefix = bp_core_get_table_prefix();
-
-			$query_placeholder = array();
-
-			$sql = ' SELECT ';
-
-			if ( $only_totalrow_count ) {
-				$sql .= ' COUNT( DISTINCT id ) ';
-			} else {
-				$sql                .= ' DISTINCT id , %s as type, post_title LIKE %s AS relevance, post_date as entry_date  ';
-				$query_placeholder[] = $this->search_type;
-				$query_placeholder[] = '%' . $search_term . '%';
-			}
+			$select      = '';
+			$where       = '';
+			$placeholder = '%' . $search_term . '%';
+			$tax         = array();
 
-			$sql .= " FROM {$wpdb->posts} p ";
+			// Get the search words from the search string.
+			$search_term_array = bb_search_get_search_keywords_by_term( $search_term, $this->pt_name );
 
-			/*
-			 ++++++++++++++++++++++++++++++++
-			 * wp_posts table fields
-			 +++++++++++++++++++++++++++++++ */
+			// Get taxonomy.
 			$taxonomies = get_object_taxonomies( $this->pt_name );
 			foreach ( $taxonomies as $taxonomy ) {
 				if ( bp_is_search_post_type_taxonomy_enable( $taxonomy, $this->pt_name ) ) {
 					$tax[] = $taxonomy;
 				}
 			}
 
-			// Tax query left join
+			$select .= ' SELECT ';
```

Comment:
Remove space after SELECT

Reply:
@jitendrabanjara1991 It's required when concat the `$select` variable